File: backend/src/main/java/space/davids_digital/sweetie/rest/dto/UserDto.java
package space.davids_digital.sweetie.rest.dto;

import java.time.ZonedDateTime;

public record UserDto(
        long vkId,
        String firstName,
        String lastName,
        long credits,
        String photoUrl,
        ZonedDateTime lastCreditGain,
        UsagePlanDto usagePlan,
        ZonedDateTime usagePlanExpiry
) {}

File: backend/src/main/java/space/davids_digital/sweetie/rest/dto/UserDto.kt
package space.davids_digital.sweetie.rest.dto

import java.time.ZonedDateTime

data class UserDto(
    val vkId: Long,
    val firstName: String?,
    val lastName: String?,
    val credits: Long,
    val photoUrl: String?,
    val lastCreditGain: ZonedDateTime?,
    val usagePlan: UsagePlanDto,
    val usagePlanExpiry: ZonedDateTime?
)
